### PR TITLE
docs: add test demonstrating Auto IAM AuthN on direct path

### LINKED ISSUE #370: Add test demonstrating direct path connections using IAM Authentication
It's possible to provide the equivalent of Auto IAM AuthN but without a connector. This feature request is for providing a database/sql driver to do that so clients don't have to wire it up themselves.

### PR DESCRIPTION
**Pull request by enocom**
Fixes #370.

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -81,19 +81,23 @@ jobs:
         uses: google-github-actions/get-secretmanager-secrets@4d6d3dfd94110800dda8d84109cb6da0f6a5919d # v1.0.1
         with:
           secrets: |-
-            ALLOYDB_URI:${{ secrets.GOOGLE_CLOUD_PROJECT }}/ALLOYDB_URI
+            ALLOYDB_INSTANCE_NAME:${{ secrets.GOOGLE_CLOUD_PROJECT }}/ALLOYDB_INSTANCE_NAME
```
Comment (jackwotherspoon) [inline .github/workflows/tests.yaml:84]: is `ALLOYDB_INSTANCE_NAME` the naming we are going forward with over URI?

File: direct_database_sql_test.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		return nil, err
+	}
+	return stdlib.GetDefaultDriver().Open(dbURI)
+}
+
+// connectDirectDatabaseSQLAutoIAMAuthN establishes a connection to your
+// database using database/sql with a small wrapper driver that provides a
+// fresh OAuth2 token for every new connection. This approach enables you to
+// connect to your AlloyDB instance directly without the AlloyDB Go Connector
+// while still using Auto IAM Authentication.
```
Comment (jackwotherspoon) [inline direct_database_sql_test.go:79]: I wonder if we should still be calling this Automatic IAM Authentication? I guess in concept to the user it is still automatic

File: direct_pgxpool_test.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	"github.com/jackc/pgx/v4/pgxpool"
+	"golang.org/x/oauth2/google"
+)
+
+// connectDirectPGXPoolAutoIAMAuthN establishes a connection to your
+// database using pgxpool and inserts a fresh OAuth2 token before initiating a
+// connection. This enables Auto IAM AuthN on the direct path without using
+// a connector.
+//
+// The function takes the revelent instance IP, a username, and a database
```
Comment (jackwotherspoon) [inline direct_pgxpool_test.go:31]: nit: typo
```suggestion
// The function takes the relevant instance IP, a username, and a database
```

**Review by jackwotherspoon — CHANGES_REQUESTED** [review]

File: direct_database_sql_test.go (hunk 2 above)
Comment (enocom) [reply direct_database_sql_test.go:79]: Introducing another term would be confusing. So I propose we call this "Automatic IAM Authentication on the Direct Path" or something similar.

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (enocom) [reply .github/workflows/tests.yaml:84]: Yes -- that's the correct term. I'm slowly fixing these everywhere.

See https://cloud.google.com/alloydb/docs/reference/rest/v1/projects.locations.clusters.instances/get for confirmation.

File: direct_database_sql_test.go (hunk 2 above)
Comment (enocom) [reply direct_database_sql_test.go:79]: WDYT?

Comment (jackwotherspoon) [reply direct_database_sql_test.go:79]: I think we may as well continue with it.

**Review by jackwotherspoon — APPROVED** [review]

**Pull request merged**